stage0/src/asm/boot.s

```diff
@@ -68,49 +68,39 @@ _protected_mode_start:
     # Switch to a flat 32-bit data segment, giving us access to all 4G of memory.
     mov $ds, %eax
     mov %eax, %ds
+    mov %eax, %es
     mov %eax, %ss
 
     # Set up a basic stack, as we may get interrupts.
     mov $stack_start, %esp
 
-    # Zero-out all the page tables: start address in EDI, value in AL, count in ECX.
-    mov $0x1000, %ecx         # each table is exactly 4096 bytes long.
-    xor %eax, %eax            # zero out eax
-    mov ${pml4}, %edi
-    rep stosb
-
-    mov ${pdpt}, %edi
-    rep stosb
-
-    mov ${pd_0}, %edi
-    rep stosb
-
-    mov ${pd_3}, %edi
-    rep stosb
-
-    mov ${pt_0}, %edi
-    rep stosb
+    # Clear BSS: base address goes to RDI, value (0) goes to AX,
+    # count/4 (we are updating 4 bytes at a time) goes into CX.
+    mov $bss_start, %edi
+    mov $bss_size, %ecx
```

Comment:
could there be a risk of leaving some garbage in the memory if BSS is not exactly divisible by 4?